### ISSUE TITLE
How to stop a call 

### ISSUE THREAD

**Issue by dtagliavini** [voryx/Thruway#248]
Hi, 
I'm using thruway with crossbar.io.

I have this problem: I call a remote procedure A from a client X (php script with thruway) to another client Y (tablet) that process received data.

If before this call, I switch the tablet Y offline, the success or error callback of the call on client X are not fired.

If this call is repeated more times nothing happens.

If I switch the tablet online, all these previous calls are delivered and callbacks are fired.

I want to abort a call after a timeout in a similar way:

```

$timer = null;
$loop = $this->getLoop();
$timer = $loop->addTimer(2, function ($timer) {
	// CANCEL REQUEST --> HOW TO DO????
	// DO ACTION
});

$session->call("it.example", [ $json ])->then+(function ($res) use ($client, $timer) {
	$client->getLoop()->cancelTimer($timer);
}, function ($error) use ($json, $structure, $campaign, $client, $timer){
	$client->getLoop()->cancelTimer($timer);
	// DO ACTION
});

```

**Comment by mbonneau** [comment]
@dtagliavini - Call cancellation is not supported in the current version of the client.

It is something that could be added by using the cancellation feature of the promises that are returned from the methods.

This would actually be a pretty good feature to add.

Right now with most new projects, I have been using an RxPHP-based client (https://github.com/voryx/RxThruwayClient) that supports cancellation.

**Comment by mbonneau** [comment]
@dtagliavini - I implemented call cancelation on the Client in the 0.5 branch.

I did see when testing against crossbar, however, that crossbar has not implemented call cancelation. (see https://github.com/crossbario/crossbar/issues/299)

Call cancelation does work with the Thruway router.

**Comment by mbonneau** [comment]
@dtagliavini - I am closing this issue. As a side note - the latest version crossbar now supports call cancelation.
